### PR TITLE
Client: don't clunk the fid after the remove call

### LINKED ISSUE #79: Tries to clunk fid after remove
The `remove` function (https://github.com/mirage/ocaml-9p/blob/cb9f5a6c83051ac5dfa7fc7a43ca6a2910a0a241/lib/protocol_9p_client.ml#L394) uses `with_fid`, which tries to clunk the fid after use: https://github.com/mirage/ocaml-9p/blob/cb9f5a6c83051ac5dfa7fc7a43ca6a2910a0a241/lib/protocol_9p_client.ml#L326 and https://github.com/mirage/ocaml-9p/blob/cb9f5a6c83051ac5dfa7fc7a43ca6a2910a0a241/lib/protocol_9p_client.ml#L326

This results in e.g.

```
+222268us fs9p       [DEBUG] C ((tag (1)) (payload (Remove ((fid 1)))))
+222276us fs9p       [DEBUG] S ((tag (1)) (payload (Remove ())))
+222300us fs9p       [DEBUG] C ((tag (1)) (payload (Clunk ((fid 1)))))
+222309us fs9p       [DEBUG] S ((tag (1))
                                (payload
                                 (Err ((ename "Unknown fid 1") (errno (0))))))
```

The spec says:

> The remove request asks the file server both to remove the file represented by fid and to clunk the fid, even if the remove fails.

### PR DESCRIPTION
**Pull request by djs55**
The spec says

> The remove request asks the file server both to remove the file
> represented by fid and to clunk the fid, even if the remove fails.

so we must not clunk the fid as well, since it may now be in use by
another operation.

Fixes #79

Signed-off-by: David Scott <dave@recoil.org>

### CODE REVIEW

File: lib/protocol_9p_client.ml [hunk 1]
```diff
@@ -395,16 +395,22 @@ module Make(Log: Protocol_9p_s.LOG)(FLOW: V1_LWT.FLOW) = struct
...
+    >>= function
+    | Error e ->
+      (* We must clunk the fid ourselves *)
+      clunk t newfid
+      >>*= fun _ ->
+      Lwt.return (Error e)
+    | Ok _ ->
+      remove t newfid
+      >>*= fun _ ->
+      (* Fid has been clunked by the remove call *)
```
Comment (talex5) [inline lib/protocol_9p_client.ml]: Don't we still need to mark the FID as free at our end?

Comment (djs55) [inline lib/protocol_9p_client.ml]: Good point, let me try to remember how this library actually works :)

**Comment by djs55** [comment]
BTW I think the `deallocate_fid` function was dubious too, see the patch for that I just added.

**Comment by talex5** [comment]
LGTM

**Pull request merged**